ocaml/xapi/message_forwarding.ml

```diff
@@ -3238,6 +3252,34 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
       Sm.assert_session_has_internal_sr_access ~__context ~sr:self;
       Local.SR.set_physical_utilisation ~__context ~self ~value
 
+    let assert_all_pbds_unplugged ~__context ~sr =
+      let any_pbds_attached_to_sr = Db.PBD.get_all_records ~__context
+      |> List.exists (fun (_ref, pbd) ->
+          pbd.API.pBD_SR = sr && pbd.API.pBD_currently_attached)
+      in
+
+      if any_pbds_attached_to_sr then
+        raise (Api_errors.Server_error(Api_errors.sr_has_pbd, [ Ref.string_of sr ]))
+
+    let assert_sr_not_indestructible ~__context ~sr =
+      let oc = Db.SR.get_other_config ~__context ~self:sr in
+
+      match List.assoc_opt "indestructible" oc with
```

Comment:
do we have a file containing consts for this sort of thing?

Reply:
Yes: xapi_globs.ml.